cpp/src/parquet/arrow/arrow-schema-test.cc

```diff
@@ -849,5 +851,21 @@ TEST(InvalidSchema, ParquetNegativeDecimalScale) {
                 ToParquetSchema(arrow_schema.get(), *properties.get(), &result_schema));
 }
 
+TEST(TestFromParquetSchema, CorruptMetadata) {
+  // PARQUET-1565: ensure that an IOError is returned when the parquet file contains
+  // corrupted metadata.
+  auto path = test::get_data_file("PARQUET-1481.parquet", false /*is_good*/);
```

Comment:
I understand that the comment `/*is_good*/` is intended to clarify that `false` is being passed as the value of the `is_good` parameter here. However, prior to explicitly double checking on the signature of `get_data_file`, I misinterpreted the meaning of this as "PARQUET-1481.parquet *is a good* file".

Since you have added similar comments to other uses of `get_data_file`, I wonder if this suggests that it might be a bit clearer to use named constants or enum values here instead, like `test::DataFileType::BAD`, `test::DataFileType::GOOD`, etc.

Using this approach might makes things more easily extensible in the future too. For example, I could imagine potentially wanting to only select test data files that belong to a particular "class" of "bad" files, like `test::DataFileType::BAD_HEADER`, `test::DataFileType::BAD_SCHEMA`, `test::DataFileType::BAD_PHYSICAL_TYPE`, etc.

Reply:
I'm wary of adding an enum implementation for a singleton edge-case.

@kevingurney, I wonder what you think of using the syntax recommended in the google style guide for [Function Argument Comments](https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments).  This is how it looks:

```cpp
auto path = test::get_data_file("PARQUET-1481.parquet",  /*is_good=*/ false);
```

which I think is more readable.

Reply:
Since commenting arguments in this way is explicitly mentioned in the Google style guide, I think it reasonable to stick with that approach. I don't think it matters too much in this case, but it does mention "As a last resort...".